### PR TITLE
Pod Volume Backup/Restore Refactor: Rename Command Parameters

### PR DESCRIPTION
**Pull request by Lyndon-Li**
Pod Volume Backup/Restore Refactor: Rename parameters in CRDs and commands to remove "Restic" word

### CODE REVIEW

**Comment by Lyndon-Li** [comment]
This PR is duplicated from #5345, please refer to #5345 for old comments.

**Review by ywk253100 — DISMISSED** [review]

File: pkg/apis/velero/v1/backup.go [hunk 1]
```diff
@@ -100,10 +100,18 @@ type BackupSpec struct {
 
 	// DefaultVolumesToRestic specifies whether restic should be used to take a
 	// backup of all pod volumes by default.
+	//
+	// Deprecated: this field is no longer used and will be removed entirely in v2.0.
```
Comment (reasonerjt) [inline pkg/apis/velero/v1/backup.go:104]: This is trivial but could you please rephrase to removed in future release, b/c we are not sure if it's v2.0 or other version.

Additionally, IMO we should also suggest user use `DefaultVolumesToFsBackup` instead.

File: pkg/controller/backup_controller_test.go [hunk 2]
```diff
@@ -342,14 +342,129 @@ func TestDefaultBackupTTL(t *testing.T) {
...
-			res := c.prepareBackupRequest(test.backup)
+			res := c.prepareBackupRequest(test.backup, logger)
 			assert.NotNil(t, res)
 			assert.Equal(t, test.expectedTTL, res.Spec.TTL)
 			assert.Equal(t, test.expectedExpiration, *res.Status.Expiration)
 		})
 	}
 }
 
+func TestDefaultVolumesToResticDeprecation(t *testing.T) {
```
Comment (reasonerjt) [inline pkg/controller/backup_controller_test.go:353]: Thank you for this!!!!

File: pkg/apis/velero/v1/backup.go (hunk 1 above)
Comment (Lyndon-Li) [reply pkg/apis/velero/v1/backup.go]: Done

**Review by reasonerjt — APPROVED** [review]

**Review by ywk253100 — APPROVED** [review]

**Pull request merged**